src/resources/MachineLearning/CaseClassificationConfiguration/CaseClassificationConfigurationInterfaces.ts

```diff
@@ -157,3 +155,51 @@ export interface CaseClassificationContentFields {
     fields: CaseClassificationContentField[];
     query: string;
 }
+
+interface CaseClassificationFieldCountAdvancedMode {
+    /**
+     * The query to use when building the model.
+     */
+    advancedQuery: string;
+}
+
+interface CaseClassificationFieldCountStandardMode {
+    /**
+     * The names of the sources that contain the cases that the model will use.
+     */
+    sources: string[];
+    /**
+     * An array of filtering conditions.
+     */
+    filterConditions?: FilterConditions[];
+    /**
+     * The name of the field that contains the value to use as the language.
+     */
+    languageField: string;
+    /**
+     * The case creation date range for the cases the model will use.
+     */
+    caseExtractionPeriod: ExtractionPeriod;
+}
+
+export type CaseClassificationFieldCountParams =
+    | CaseClassificationFieldCountAdvancedMode
+    | CaseClassificationFieldCountStandardMode;
```

Comment:
Are you trying to get one or the other? The way it's defined right now it's a join of both so you can have `advancedQuery` and `languageField` [without any error](https://www.typescriptlang.org/play/?#code/JYOwLgpgTgZghgYwgAgMJwM4VQG0x4GYBOMYAexADFgIcATVcgV3AEF6A3OEJegWXL0UAbwBQyScjhcefAIrNoATwBcyDGCigA5gG4xAXzFjQkWIhTosufIWKkK1WgyaswAZTA96cKAKFRCSkMFigkDHVNbRAdAG0AXQMpZCIccyYQemAySgwAfnUeZUS9ZAB6cukQZVTyKA1gAFsABxxiHOVgyTxY5jgdCBo6eiitXWSpEiwAUQAPLURckAAFaApR5BFDMsrkCFawWvIAIwArCAQwOoaCVvaETqMTCDmW+uujlqtMbDwMAhEEjLYauFjgFZ+OBNDDIAC83WQAB80L9bAD7MCnKDGOCwBxuLwIAFhIiUdY-nYgY5KDi3OAvD4-CSIAYxAg8tc2OoKejAQ4QS5ce5IVBobC4VsZISFEooGpkAByOCKgA0yF6On6gxx6kVJ0VhgqVRA5H2UCg9WQlGQYAAFih2iAIGIgA)

You can [differentiate the types this way](https://www.typescriptlang.org/play/?#code/JYOwLgpgTgZghgYwgAgMJwM4VQG0x4GYBOMYAexADFgIcATVcgV3AEF6A3OEJegWXL0UAbwBQyScjhcefAIrNoATwBcyDGCigA5gG4JUw5ID0J5GAAWwDNJw5yAdwvKADhAwJtrsBfLJ6QhhoCHBgUhQAIwgwRwhQi0sUMDcPYw0WKCQMAH51EAhOaAMpZCIcSCgmEECySlz8wuL0vBAdZjgdCBo6ejzkAqKoEqkSLABRAA8tRDqQAAVoCj7GoYMAXzExUEr4JDRMbDwMAiISOZ6GJlYwAGUwHno4KAEhUXSMTOz1TW02gG0ALojSTlSrVWoUEANaQgZRAvTIMyw5RlchQDTAAC2rhwxGAKRaPHanW6tAYPy0uhByDGECmMwQc0W2iE6hE60RyIgOJSyHIkQAVhAmWiMQQcXiEATlFtSsirDY7A5nCl3J5vL4wP5AjBglBQmQIshorF4iBEslUhh0jJuLwIPRFCp+oNmpsxBBJq50VrUgcsLh8IRiKQoZdGCxwPNnnAsbYALzpAA+AaOwbOYcoEeu7FkDtewhTaaDJxD53D5MjN3uj2ehYgBjECHqvjY6nQgeOp1DFyrubAMagccTyBEdrkjudUDUyAA5HA5wAaZCtEldCPqOeROfrJHmHjIaBQdH8i1WFB4gorzBH70iyD0LYt6G+ABCHcOpZ7Fez-ajg6xvGyAJmOE4FtOs4Lru+4DP4x6npQlqrqAEA3rYXruEyjrNq2aCfl2Ga9pWvQDkOI4gWOnzMFkHjqP80GAiua4dBuVZbjuK50gyw5MlCLLLOy6x7siIDwVAJ4YkhF4ode0gYfe2FPrh0LkDgEAAHQODoAAUbArm+K6oAAlAYQA)

Reply:
Ahh yeah :laughing: the fact that the autocomplete wasn't suggesting more parameters misled me to think that it was indeed rejecting it - I implemented your solution :+1: